File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/9 - 20 and 21 April 2019/05. Easter Eggs.py
count_eggs = int(input())

red = 0
orange = 0
blue = 0
green = 0

for egg in range(count_eggs):
    color = input()
    if color == "red":
        red += 1
    if color == "orange":
        orange += 1
    if color == "blue":
        blue += 1
    if color == "green":
        green += 1

colors = {'red': red, 'orange': orange, 'blue': blue, 'green': green}
max_colors_name = max(colors, key=colors.get)

colors_count = {red, orange, blue, green}
max_colors_count = max(colors_count)

print(f"Red eggs: {red}")
print(f"Orange eggs: {orange}")
print(f"Blue eggs: {blue}")
print(f"Green eggs: {green}")
print(f"Max eggs: {max_colors_count} -> {max_colors_name}")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/01. Online Exam - 20 and 21 April 2019/05. Easter Eggs.js
function easterEggs(input) {
    // Mask
    let colors = { 
        'red': 0,
        'orange': 0,
        'blue': 0,
        'green': 0
    };

    let maxColor = undefined;
    let maxCount = 0;

    let count = Number(input.shift());

    for (let time = 1; time <= count; time += 1) {
        let color = input.shift();
        colors[color] += 1;
        if (colors[color] > maxCount) {
            maxCount = colors[color];
            maxColor = color;
        }
    }

    console.log(`Red eggs: ${colors['red']}`);
    console.log(`Orange eggs: ${colors['orange']}`);
    console.log(`Blue eggs: ${colors['blue']}`);
    console.log(`Green eggs: ${colors['green']}`);
    console.log(`Max eggs: ${maxCount} -> ${maxColor}`)
}

easterEggs([7, 'orange', 'blue', 'green', 'green', 'blue', 'red', 'green']);  // Should return:
// Red eggs: 1
// Orange eggs: 1
// Blue eggs: 2
// Green eggs: 3
// Max eggs: 3 -> green

easterEggs([4, 'blue', 'red', 'blue', 'orange']);  // Should return:
// Red eggs: 1
// Orange eggs: 1
// Blue eggs: 2
// Green eggs: 0
// Max eggs: 2 -> blue
